wagtail/admin/templates/wagtailadmin/base.html

```diff
@@ -8,9 +8,9 @@
     <main class="content-wrapper w-overflow-x-hidden" id="main">
         <div class="content">
             {# Always show messages div so it can be appended to by JS #}
-            <div class="messages" role="status">
-                {% if messages %}
-                    <ul>
+            <div class="messages" role="status" data-controller="w-messages" data-action="w-messages:add@document->w-messages#add" data-w-messages-new-message-class="new" data-w-messages-show-class="appear">
```

Comment:
might be nice to wrap these attributes to multiple lines, this is getting a bit long and it will be easier to read.